pep-9999.rst

```diff
@@ -81,9 +81,214 @@ includes stubs for Python's standard library and several third-party
+operator ``...`` on the same line as the closing parenthesis and colon.
+Do not include docstrings.
+
+Yes::
+
+    def to_int1(x: str) -> int: ...
+    def to_int2(
+        x: str,
+    ) -> int: ...
+
+No::
+
+    def to_int1(x: str) -> int:
+        return int(x)
+    def to_int2(x: str) -> int:
+        ...
+    def to_int3(x: str) -> int: pass
+
+Use the ellipsis operator ``...`` in place of actual default argument
+values. Use an explicit ``Optional`` annotation instead of
+a ``None`` default.
+
+Yes::
+
+    def foo(x: int = ...) -> None: ...
+    def bar(y: Optional[str] = ...) -> None: ...
+
+No::
+
+    def foo(x: int = 0) -> None: ...
+    def bar(y: str = None) -> None: ...
+    def baz(z: Optional[str] = None) -> None: ...
+
+Private Classes
+---------------
+
+If a function or method returns an instance of a private class, the
+type stub should provide a public, stub-only class::
+
+    # Only available during type checking
+    class SomeData:
+        x: int
+        def foo() -> None: ...
+
+    def get_some_data() -> SomeData: ...
+
+Language Features
+-----------------
+
+Use the latest language features available as outlined
+in the Syntax_ section, even for stubs targeting older Python versions.
+Do not use quotes around forward references and do not use ``__future__``
+imports.
+
+Yes::
+
+    class Py35Class:
+        x: int
+        forward_reference: OtherClass
+    class OtherClass: ...
+
+No::
+
+    class Py35Class:
+        x = 0  # type: int
+        forward_reference: 'OtherClass'
+    class OtherClass: ...
+
+Types
+-----
+
+Use ``float`` instead of ``Union[int, float]``. For argument types,
+use ``bytes`` instead of ``Union[bytes, memoryview, bytearray]``.
+
+Use ``Text`` in stubs that support Python 2 when something accepts both
+``str`` and ``unicode``. Avoid using ``Text`` in stubs or branches for
+Python 3 only.
+
+Yes::
+
+    if sys.version_info < (3,):
+        def foo(s: Text) -> None: ...
+    else:
+        def foo(s: str, *, i: int) -> None: ...
+    def bar(s: Text) -> None: ...
+
+No::
+
+    if sys.version_info < (3,):
+        def foo(s: unicode) -> None: ...
+    else:
+        def foo(s: Text, *, i: int) -> None: ...
+
+Prefer protocols and abstract types (``Mapping``, ``Sequence``, ``Iterable``,
+etc.) for arguments. Prefer concrete types (``List``, ``Dict``, etc.) for
+return values of concrete implementations. The return values of protocols
+and abstract base classes must be judged on a case-by-case basis.
+
+Yes::
+
+    def map_it(input: Iterable[str]) -> List[int]: ...
+    def create_map() -> Dict[str, int]: ...
+
+No::
+
+    def map_it(input: List[str]) -> List[int]: ...
+    def create_map() -> MutableMapping[str, int]: ...
+
+Maybe::
+
+    class MyProto(Protocol):
+        def foo(self) -> List[int]: ...
+        def bar(self) -> Mapping[str]: ...
+
+Avoid ``Union`` return types, since they require ``isinstance()`` checks.
+Use ``Any`` if necessary.
```

Comment:
What about #18? Does this interact at all with mypy's `Any` checks?

Reply:
True, the result of #18 needs to be added still. I would like to do that in a separate pull request, as it might require a bit of discussion about the exact wording.

Reply:
That's fine with me.